app/models/entry_importer.rb

```diff
@@ -0,0 +1,169 @@
+class EntryImporter
+  attr_reader :current_notebook, :root_path
+  def initialize(current_notebook)
+    @current_notebook = current_notebook
+    @root_path = current_notebook.import_path
+  end
+
+  def resolve_and_import!(identifier)
+
+    # case identifier
+    # when "stylesheets/application"
+      # return render_stylesheet
+    # end
+    # TODO: replace hacky & brittle with more well-defined search
+    # i.e. exact lookups vs just fuzzy searching
+    # ALSO: need to think about how this would interact with scss templates.
+    file_path = build_file_path(identifier)
+    glob_path = file_path + "*"
+    file_path = Dir[glob_path].select { |f| f =~ /#{file_path}(\.html|\.md|\.markdown)+/ }.reject {|f| File.directory?(f) }.first
+    if file_path && looks_like_text?(file_path)
+      # TODO: hrm, should the identifier come from the request? this will blow up somehow
+      import!(identifier, file_path)
+    else
+      puts "couldn't find nothin' to import"
+      return nil
+    end
+  end
+
+  def import!(identifier, file_path)
+    entry_attributes = entry_attributes_from_markdown(identifier, file_path)
+
+    entry = add_entry!(entry_attributes)
+  end
+
+  def looks_like_text?(file_path)
+    file_path =~ /\.(md|markdown|html|erb)$/
+  end
+
+  def build_file_path(path_info)
+    clean_path_info = Rack::Utils.clean_path_info(path_info)
+    ::File.join(@root_path, clean_path_info)
+  end
+
+  def entry_attributes_from_markdown(identifier, md_path)
+    loader = FrontMatterParser::Loader::Yaml.new(allowlist_classes: [Time, Date, DateTime])
+    md_parser = FrontMatterParser::SyntaxParser::Md.new
+    parsed_file = FrontMatterParser::Parser.new(md_parser, loader: loader).call(File.read(md_path))
+
+    occurred_at = parsed_file["occurred_at"]
+    if occurred_at.blank?
+      # let's try to guess it from the file
+      basename = File.basename(md_path)
+      date = basename.match(/([0-9]{4}-*[0-9]{2}-*[0-9]{2}-*)/).to_a[0]
+
+      # most of the time this does the right thing but it does have the habit
+      # of sometimes throwing an exception
+      begin
+        occurred_at = DateTime.parse(date.to_s)
+      rescue ArgumentError
+      end
+
+      # if still nil, let's look at the file itself
+      if occurred_at.nil?
+        occurred_at = File.ctime(md_path)
+      end
+    end
+
+    # TODO: DON'T DO THIS GIT DOESN"T SET CTIME!!!!!!
+    created_at = parsed_file["created_at"] || File.ctime(md_path)
+    updated_at = parsed_file["updated_at"] || File.mtime(md_path)
```

Comment:
this actually has caused all sorts of annoying issues when regenerating the site, cos the updated at will reflect the disk mtime & not the last time the file was actually modified.